### DISCUSSION TITLE
Can Gravitino replace HMS? Similar to AWS Glue, can it replace HMS, because it is compatible with HMS client?

### DISCUSSION THREAD

**Discussion by melin** [apache/gravitino#4327] [Q&A]
gravitino 能够替代hms？类似aws glue 可以替代hms，只是兼容hms client

**Comment by xunliu** [comment]
hi @melin This is a very interesting thing, Do you want to implement this future?
additionally, Please change this discussion title and context to English, Thanks!

**Comment by shaofengshi** [comment]
I think this is a good question, we should add it to FAQ or somewhere similar for new Gravitino users.

As Gravitino is an Apache incubator project now, which is open for people all over the wold, so English is prefered. I'd like to translate the question to English before answering it.

**Comment by shaofengshi** [comment]
Gravitino is an open data catalog for variaous data sources, including relational database, data warehouse (including Apache Hive), lake house formats (iceberg, hudi, paimon, etc), as well as streaming data and unstructuerd data. So it's purpose and position is different from HMS as well as Glue catalog. Gravitino will have it's own REST APIs, instead of be compitable with HMS's APIs. 

To support engines connect with Gravitino, we build connctors, such as Trino connector, Apache Spark connector, and the Apache Flink connector is also on the way.

**Comment by melin** [comment]
What users create with spark are parquet and orc tables, Gravitino supports?

If Gravitino is used, hms is introduced again in order to store metadata for the parquet and orc tables. Two metadata systems need to be maintained.
